### PR TITLE
P8- MoE Architecture : Issue-141 : Add MoE cadence validation and risk decision framework

### LINKED ISSUE #17: Final (PreTraining) Benchmarks
# Team 17: Final Benchmarks & Evaluation Strategy (Pretraining Truth Gate)

## Category

**Evaluation / Meta**

---

## Objective (Brief)

Produce a **single, clean, defensible evaluation of the pretrained model**, while also **mapping the full benchmark landscape** required for downstream SFT, alignment, and agentic evaluation.

Team 17 owns **benchmark selection integrity** and **final pretraining evaluation execution**.

This team does **not** optimize models, tune prompts, or interpret results beyond their valid scope.

---

## Purpose (Charter)

To ensure that:

* pretraining progress is measured **honestly**
* claims are **phase-correct**
* downstream teams inherit a **clear evaluation roadmap**
* no benchmark is misused or prematurely reported

This team prevents the most damaging failure mode in research systems:

> **Running the wrong benchmark at the wrong time and drawing the wrong conclusion.**

---

## Role Definition (Authoritative)

Team 17 has **two distinct, non-overlapping roles**.

---

### Purpose

To **identify, classify, and document all relevant benchmarks** across the **entire model lifecycle**, including those **not run in this phase**.

This role informs:

* dataset choices upstream
* SFT and alignment planning (Team 18 / Team 19)
* future claims and papers

---

### What You Must Do (Role A)

1. Survey and catalog benchmarks across categories:

   * pretraining (base models)
   * instruction following
   * reasoning
   * code
   * multilingual / Indic
   * agentic behavior
   * safety & alignment

2. For each benchmark, document:

   * what it actually measures
   * which training phase it is valid for
   * whether it assumes instruction tuning
   * quantization tolerance
   * known pitfalls / misuse risks

3. Explicitly classify benchmarks as:

   * **RUN NOW (pretraining-valid)**
   * **DEFERRED (post-SFT / alignment)**
   * **RESEARCH-ONLY / UNSAFE**

---

### Hard Rule (Role A)

Team 17 **does not execute** deferred benchmarks.

Discovery ≠ execution.

---

### Purpose

To run **one final, clean evaluation pass** that measures **representational competence**, not behavior.

This is the **last evaluation before SFT begins**.

---

### Benchmark Eligibility Rules (Non-Negotiable)

Benchmarks executed in this role must:

1. Be valid for **base (non-instruction) models**
2. Not require chat formatting
3. Not require alignment or preference learning
4. Be prompt-minimal or continuation-based
5. Be quantization-tolerant (INT8 / INT4)
6. Be comparable across checkpoints

Anything violating these rules is **deferred**.

---

## Authorized Benchmark Set (Suggestive)

Team 17 is authorized to run **only the following benchmarks** for pretraining evaluation:

### 1) **MMLU (Full, Quantized)**

* Run once on the final pretrained checkpoint
* Report **category-wise results only**
* No instruction prompting
* Framed strictly as:

  > *“Knowledge and reasoning substrate evaluation”*

---

### 2) **GSM8K (Continuation-Style Only)**

* No chain-of-thought prompting
* No explanations
* Measures latent arithmetic structure, not reasoning skill

---

### 3) **Code Completion Evaluation (HumanEval-Style, Modified)**

* Completion-only
* No docstring instructions
* No natural language explanations
* Measures code structure emergence

---

### 4) **Indic / Multilingual Continuation Probes**

* Perplexity or continuation-based
* No translation or QA
* Presence and stability only, not fluency claims

---

### 5) *(Optional)* **Minimal Pair / Grammar Sensitivity (BLiMP-style)**

* Used as a representation health signal
* Optional, but strongly recommended if time permits

---

## Explicitly Disallowed in This Phase

The following **must not be run or reported** by Team 17:

* Instruction-tuned MMLU variants
* MT-Bench, AlpacaEval, Arena
* Tool-use benchmarks
* Agent planning benchmarks (WebArena, SWE-bench, etc.)
* Safety / refusal benchmarks
* Preference modeling benchmarks

These are deferred to **Team 18 / Team 19**.

---

### A) Evaluation Integrity

* No cherry-picking
* No prompt tuning to inflate scores
* No cross-phase comparisons

---

### B) Claim Discipline

* Results must be labeled **“Pretraining-Only”**
* No behavioral or instruction-following claims
* No comparisons against instruction-tuned models

---

### C) Upstream Feedback Loop

* Flag weak areas to:

  * data teams
  * curriculum teams
  * synthetic data teams
* Do not propose fixes; only surface signals

---

## Inputs

* Final pretrained checkpoint(s) from **Team 12**
* Evaluation harnesses and reduced eval logic from **Team 16**
* Tokenizer artifacts from **Team 6**
* Quantization constraints

---

## Tools

* Quantization tools (INT8 / INT4)
* Evaluation harnesses
* Reporting and plotting scripts

---

## You Must Submit (Brief)

1. **Benchmark Landscape Report**

   * full catalog (run + deferred)
   * phase-correct classification

2. **Final Pretraining Benchmark Report**

   * raw scores
   * category-wise breakdowns
   * trend context (if available)

3. **Methodology Notes**

   * prompt formats
   * quantization details
   * reproducibility instructions

---

## Outputs (Charter)

* Credible, phase-correct pretraining evaluation
* Benchmark roadmap for downstream SFT/alignment
* Data-driven signals for upstream adjustments

---

## Success Criteria

* Results are credible and defensible
* No benchmark misuse
* Downstream teams trust the evaluation
* No overclaiming in reports or papers

---

## Failure Conditions

* Instruction-only benchmarks run on base models
* Ambiguous or inflated claims
* Results that cannot be reproduced
* Confusion between pretraining and alignment performance

---

## Depends on (Blocking Prerequisites)

**Teams 16 and 12**
(and final pretrained model availability)

---

## Blocks (Downstream Teams)

**Team 18**

---

### Setup / Prep

**Feb 11 – Feb 12**

* Finalize benchmark list
* Lock evaluation methodology

### Peak

**Feb 12 – Feb 16**

* Run authorized benchmarks
* Produce reports

### Monitor / Support

—

### Done / Frozen

**Feb 17**

* Results frozen
* Reports archived
* Handoff complete

### LINKED ISSUE #141: 8.2: Paper Findings & Component De-Risking
**Objective**: Scientifically justify the growth cadence and validate high-risk components (MTP/mHC).

- Cadence Validation: Formally evaluate the 1B (Dense) -> 3B (MoE) -> 8B (Dense) -> 70B (MoE) growth path.
- MTP Research: Review DeepSeek/Meta findings on Multi-Token Prediction. Does it aid "routing learning" in the 3B stage? (Decision: Go/No-Go).
-  mHC Research: Review stability analysis for mHC vs. standard Residuals. Does mHC vanish/explode gradients in deep networks (70B)? (Decision: Go/No-Go).
- Null Expert Review: Analyze ArXiv 2601.11616 to confirm "Null" implementation details.

### PR DESCRIPTION
**Pull request by Sualeh77**
- Finalize cadence_validation.md (Dense → MoE-small → Dense-deep → MoE-large)
- Lock risk_decisions.md (null experts, MTP NO-GO, mHC GO@70B)
- Add paper_findings_matrix.md with cross-paper MoE scaling analysis
- Ground expert count recommendations per stage from literature

# Pull Request Template

## Description

Please include a summary of the changes and the related issue. Highlight any key points that reviewers should focus on.

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if applicable).
- [ ] My code follows the style guidelines, gitflow branching strategy, and naming conventions of this project [Contribution Guidelines](<https://github.com/The-School-of-AI/LLM/tree/main/experiments/>

## Reviewers

- [ ] **Reviewer 1**: A member from your own team.
- [ ] **Reviewer 2**: A member from the repo owners team (@The-School-of-AI/llm-repo-owners).

> **Note**: Every pull request requires atleast 2 reviewers/approvers before it can be merged.

### CODE REVIEW

File: experiments/8_moe_architecture/paper_findings_matrix.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+- [ ] Dense warmup (20k steps or similar) before enabling MoE and null routing.
+- [ ] Router expansion to N + M logits (duplicate null logit M times). Follow Algorithm 1 style modifications.
+- [ ] Keep `k_max = 16` when `rho = 0.5` to preserve `E[K_real] = 8`.
+- [ ] Use bias-only (aux-loss-free) balancing where DeepSeek uses it for large-scale runs; with null experts, apply standard global load balancing over N + M slots but tune weight carefully (paper used `L_bal` weight ~2e-2 and z-loss ~1e-3 as a starting point).
+- [ ] Monitor: zero-compute token ratio, per-expert token counts, router entropy, and downstream eval metrics to detect collapse or polarization.
+
+---
+
+## Consolidated final table (all stages)
```
Comment (rajankumar-spec) [inline experiments/8_moe_architecture/paper_findings_matrix.md]: Is this Correct? I see some discrepancy from above .

**Review by rajankumar-spec — CHANGES_REQUESTED** [review]
Please review the comment.

**Review by rajankumar-spec — APPROVED** [review]

**Review by mohantee — APPROVED** [review]

**Review by abi2024 — APPROVED** [review]

**Comment by pankajkr23** [comment]
Please reopen with staging if still needed
